src/v/cluster/rm_stm.cc

```diff
@@ -2808,6 +2829,9 @@ rm_stm::load_abort_snapshot(abort_index index) {
       meta_parser.bytes_left());
 
     auto data_buf = co_await read_iobuf_exactly(reader->input(), snapshot_size);
+    uint64_t abort_snapshot_disk_size = co_await reader->get_snapshot_size();
+    _abort_snapshot_sizes.emplace(
+      std::make_pair(index.first, index.last), abort_snapshot_disk_size);
```

Comment:
I don't think this is needed anymore. Basically we only need to ensure that `_abort_snapshot_sizes` map is in sync with `_log_state.abort_indexes` and I think it is not changed when this function is called.